src/Controls/src/Core/UriImageSource.cs

```diff
@@ -113,11 +126,20 @@ async Task<Stream> DownloadStreamAsync(Uri uri, CancellationToken cancellationTo
 		{
 			try
 			{
-				using var client = new HttpClient();
```

Comment:
These things need to be actually checked. Caching and downloading is partially done in the image source services, so maybe those should take an additional ctor parameter of an http client factory? We probably should mark this GetStream obsolete and not use it as it appears to do things but in fact is just a small way to download the image.

Reply:
In the review comment @mattleibow also stated:

> This request changes is not for the PR but rather we need to check and see if we should be doing anything here at all. For backwards compat, we can keep this and remove later, but most of the work is in the services so doing this here now adds code here that is not the same in the services.

Should we log a bug for that? Is there any action to take in this PR regarding that?

Reply:
I was meaning that we need to decide on the desired action here. We _can_ maybe add code, but then we change behaviour, and then if we decide we want to do the work in the services we now either have to maintain 2 things or revert the change.

Reply:
Sorry not sure I follow. What needs to be checked? It seems to be that in the default case (e.g. update from MAUI 7 to MAUI 8), there would be no behavior change at all, no? Both before and after, a new HttpClient would be created each time, used, and then disposed. Of course, with this change, users can register a custom service and do something different, but that's the desired effect of this change.

Reply:
I am meaning we are making this method do something different, but I am suggesting we do away with this altogether. Image loading is done via the image source services. Android passes the url to glide and never even calls this. So when you do things here, it will not take effect on Android - same with Tizen. 

My overall thing here is that making this change is not going to have the desired effect, so adding all the new types and docs is incorrect.

Adding headers on Android with glide is an entirely different thing and needs to call glide APIs.

Reply:
I didn't realize Android and Tizen had different code paths. Thanks for the headsup. If this doesn't work the same on all platforms it wouldn't really make sense to do.

Reply:
@mattleibow said:

> Image loading is done via the image source services

Hmm that surprises me. Then what is the responsibility of this type? It gets used, but only sometimes, on some platforms, subject to conditions?

Should this type gain _more_ responsibilities and have more smarts for each platform? Or should it be largely whittled down to a bare minimum and delegates all responsibility to the platforms?

Do you have thoughts on what strategy could be employed to enable the generic feature of "I want to customize how image sources are loaded across all platforms"?

Reply:
Maybe it makes sense to solve it per use case. E.g. add the following API:
```csharp
            builder
                .UseMauiApp<App>()
                .ConfigureMauiHttpRequests(http =>
                {
                    http.Timeout = TimeSpan.FromSeconds(60);
                    http.DefaultRequestHeaders.Authorization = /* whatever */;
                });
```

And then it decided per platform what to do with those settings -  whether to pass it to Glide or to use during HttpClient creation.

Reply:
@Dreamescaper yeah that could certainly work, though of course we'd have to figure out what each platform actually supports 🥴

Reply:
Sure. That's why I suggest I suggest to handle requested use cases, not to try to handle every possible permutation that HttpClient allows. E.g. start with headers only, and expand when requested.
If the use case is very common - _probably_ the underlying platform supports it.

Reply:
Good idea. Sound like a more doable way to achieve the desired outcome.